app/v1/connectors/BaseNrsConnector.scala

```diff
@@ -39,23 +40,22 @@ trait BaseNrsConnector {
       "User-Agent" -> appConfig.appName
     )
 
-  def nrsPost[Body: Writes, Resp](body: Body, uri: NrsUri[Resp], defaultResult: NrsOutcome[Resp])(implicit ec: ExecutionContext,
-                                                                                      hc: HeaderCarrier,
-                                                                                      wsReads: WsReads[NrsOutcome[Resp]]): Future[NrsOutcome[Resp]] = {
+  def nrsPost[Body: Writes, Resp](body: Body, uri: NrsUri[Resp], submissionId: String)(implicit ec: ExecutionContext,
+                                                                 hc: HeaderCarrier,
+                                                                 wsReads: WsReads[NrsOutcome[NrsResponse]]): Future[NrsOutcome[NrsResponse]] = {
 
-    def doPost(implicit hc: HeaderCarrier): Future[NrsOutcome[Resp]] = {
+    def doPost(implicit hc: HeaderCarrier): Future[NrsOutcome[NrsResponse]] = {
 
       ws.url(s"${appConfig.nrsBaseUrl}/${uri.value}")
         .withHttpHeaders(hc.headers: _*)
         .withRequestTimeout(appConfig.nrsMaxTimeout)
         .post(Json.toJson(body))
-        .map(wsReads.wsRead(_, defaultResult)).recover {
+        .map(res => wsReads.wsRead(res, Right(NrsResponse.empty)).map(_.copy(nrSubmissionId = submissionId))).recover {
```

Comment:
Would this not work?

```suggestion
        .map(res => wsReads.wsRead(res, Right(NrsResponse.empty.copy(nrSubmissionId = submissionId)))).recover {
```